docs/integration/camel-dhis2-component.md

```diff
@@ -0,0 +1,308 @@
+---
+title: Camel DHIS2 Component
+sidebar_label: Camel DHIS2 Component
+slug: "/integration/camel-dhis2-component"
+---
+
+The Camel DHIS2 Component leverages the DHIS2 Java SDK to integrate Apache Camel with DHIS2. The component is distributed with Apache Camel and can be added to a Maven project POM like so:
+
+```xml
+<dependency>
+    <groupId>org.apache.camel</groupId>
+    <artifactId>camel-dhis2</artifactId>
+    <version>x.x.x</version>
+    <!-- use the same version as your Camel core version -->
+</dependency>
+```
+
+## Getting started
+
+:::note
+A basic understanding of Apache Camel and endpoints is assumed. Visit the [Apache Camel page on the DHIS2 Developer Portal](/docs/integration/apache-camel) to learn the fundamental Camel concepts. All Camel route examples given in this page are expressed in the Java-based DSL.
+:::
+
+A DHIS2 endpoint is declared in a route using the following URI format:
+
+```
+dhis2:apiName/methodName?[parameters]
+```
+
+### Endpoint parameters
+
+Parameters common across all APIs are listed below:
+
+#### baseApiUrl
+
+DHIS2 server base API URL to sends requests to.
+
+#### username
+
+Username of the DHIS2 user to operate as.
+
+#### password
+
+Password of the DHIS2 username.
+
+#### pat
+
+Personal access token to authenticate with DHIS2. This option is mutually exclusive to `username` and `password`
+
+#### client
+
+Reference to the underlying client communicating with the DHIS2 server. Referencing a client avoids the need to duplicate the base API URL and credentials when declaring multiple DHIS2 endpoints. Duplicating the base API URL and credentials is negatively impacts run-time performance because an identical DHIS2 Java SDK client is created for each endpoint. This option is mutually exclusive to `baseApiUrl`, `username`, `password`, and `pat`. The subsequent example demonstrates how to (1) create the client, (2) register it with Camel, and then (3) reference the client from a DHIS2 endpoint:
+
+```java
+...
+...
+
+public class MyRouteBuilder extends RouteBuilder {
+
+    public void configure() {
+        org.hisp.dhis.integration.sdk.api.Dhis2Client dhis2Client = org.hisp.dhis.integration.sdk.Dhis2ClientBuilder.newClient("https://play.dhis2.org/40.2.2/api", "admin", "district").build();
+        getCamelContext().getRegistry().bind("dhis2Client", dhis2Client);
+
+        from("direct:resourceTablesAnalytics")
+            .to("dhis2://resourceTables/analytics?skipAggregate=true&skipEvents=true&lastYears=1&client=#dhis2Client");
+    }
+}
+```
+
+:::note
+Visit the [DHIS2 Java SDK page on the DHIS2 Developer Portal](/docs/integration/dhis2-java-sdk) to learn more about the DHIS2 Java SDK.
+:::
+
+The `client` parameter in the above route snippet is set to `#dhis2Client`. `#dhis2Client` is an entry in Camel registry which is bound to an instance of `Dhis2Client` thanks to the following line of code:
+
+`getCamelContext().getRegistry().bind("dhis2Client", dhis2Client);`
```

Comment:
There are several blocks in this file that are with single-ticks, instead of code-blocks. Could you convert them all to code-blocks? 

Also in all the endpoints/methods a lot of code-blocks that are with single ticks